CMakeLists.txt

```diff
@@ -1,6 +1,39 @@
 cmake_minimum_required(VERSION 3.23.1)
 project(flashinfer CUDA CXX)
 
+option(ENABLE_TESTING "Enable testing" ON)
+
+# Everything in here should NOT happen when building the package for distribution.
+if (ENABLE_TESTING)
+
+  # Generates the compile_commands.json file, which is used by the language server (clangd) to do static analysis.
+  set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+  # By default, CUDA uses response files to pass arguments to the compiler. This is not well supported by tooling, such as clangd.
+  set(CMAKE_CUDA_USE_RESPONSE_FILE_FOR_INCLUDES 0)
+  set(CMAKE_CUDA_USE_RESPONSE_FILE_FOR_LIBRARIES 0)
+  set(CMAKE_CUDA_USE_RESPONSE_FILE_FOR_OBJECTS 0)
+
+  enable_testing()
+
+  # Add lineinfo to the nvcc flags only in release builds.
```

Comment:
Why do we do that?
(the comment is basically describing what I can already read on the next line, but does not provide more information here.)

Reply:
Added more detailed doc. This allows source mapping in ncu to map assembly back to code and identify bottlenecks.